sympy/sets/sets.py

```diff
@@ -1551,6 +1585,15 @@ def is_iterable(self):
         if self.args[0].is_iterable:
             return True
 
+    @property
+    def is_finite_set(self):
```

Comment:
refactor the logic to
```
return a_finite and B.is_finite_set
```

Reply:
This is using fuzzy logic with True, False or None. Simplifying it in the way that you suggest will not do the correct thing when either `A.is_finite` and `B.is_finite` give `None` rather than `True/False`.